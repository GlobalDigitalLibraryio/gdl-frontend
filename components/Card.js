import React from 'react';

const Card = ({ children, ...props }) => (
  <div {...props}>
    {children}
    <style jsx>{`
      div {
        background-color: white;
        box-shadow: 0 2px 3px rgba(10, 10, 10, 0.1), 0 0 0 1px rgba(10, 10, 10, 0.1);
        color: #4a4a4a;
        max-width: 100%;
        position: relative;
      }
    `}</style>
  </div>
);

const CardContent = ({ children, ...props }) => (
  <div {...props}>
    {children}
    <style jsx>{`
      div {
        padding: 1.5rem;
      }
    `}</style>
  </div>
);

Card.Content = CardContent;

export default Card;
