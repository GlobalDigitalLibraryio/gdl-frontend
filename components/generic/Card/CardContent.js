/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import styled from 'styled-components';

export default styled.div`
  padding: 1.5rem;
`;
