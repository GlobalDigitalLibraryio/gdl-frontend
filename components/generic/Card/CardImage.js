/**
 * Copyright (c) 2017-present, Global Digital Library.
 * 
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 * 
 */

import styled from 'styled-components';

export default styled.div`
  display: block;
  position: relative;
`;
