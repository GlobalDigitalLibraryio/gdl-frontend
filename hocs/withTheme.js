// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import * as React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

/* eslint-disable react/no-multi-comp */

const theme = {
  primaries: {
    primary: '#20588f',
    secondary: '#507aa4',
    tertiary: '#a5bcd3',
    light: '#ceddea',
    highlight: '#edf7ff',
    dark: '#184673',
  },
  supports: {
    greenPrimary: '#5cbc80',
    greenHighlight: '#edfff4',
    greenDark: '#359258',
  },
  grays: {
    dark: '#444',
    white: '#fff',
    steel: '#666',
    jumbo: '#888',
    silverSand: '#bbb',
    platinum: '#e3e3e3',
    gallery: '#eff0f2',
    desertStorm: '#f8f8f8',
  },
};

class Theme extends React.Component<{ children: React.Node }> {
  static childContextTypes = {
    reactIconBase: PropTypes.object,
  };

  getChildContext() {
    return {
      reactIconBase: {
        size: 24,
      },
    };
  }

  render() {
    return <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>;
  }
}

/**
 * HoC that wraps a page with Theme
 */
function withTheme(Page: React.ComponentType<any>) {
  return class PageWithTheme extends React.Component<any> {
    static getInitialProps(context) {
      if (typeof Page.getInitialProps === 'function') {
        return Page.getInitialProps(context);
      }
      return undefined;
    }

    render() {
      return (
        <Theme>
          <Page {...this.props} />
        </Theme>
      );
    }
  };
}

export { Theme, theme, withTheme as default };
