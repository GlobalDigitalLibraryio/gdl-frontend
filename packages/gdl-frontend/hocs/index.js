// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2018 GDL
 *
 * See LICENSE
 */

export { default as withErrorPage } from './withErrorPage';
export { default as securePage } from './securePage';
