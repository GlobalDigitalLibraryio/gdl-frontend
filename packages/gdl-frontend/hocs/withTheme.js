// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import * as React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'emotion-theming';
import { MuiThemeProvider } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import JssProvider from 'react-jss/lib/JssProvider';

import getPageContext from '../getPageContext';
import type { Context } from '../types';
import { misc } from '../style/theme';

const theme = { space: [], breakpoints: misc.breakpoints };

class Theme extends React.Component<{ children: React.Node }> {
  static childContextTypes = {
    reactIconBase: PropTypes.object
  };

  getChildContext() {
    return {
      reactIconBase: {
        size: misc.iconSize
      }
    };
  }

  render() {
    return <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>;
  }
}

//const jss = create(jssPreset());

/**
 * HoC that wraps a page with Theme
 */
function withTheme(Page: React.ComponentType<any>) {
  return class PageWithTheme extends React.Component<any> {
    static getInitialProps(context: Context) {
      if (typeof Page.getInitialProps === 'function') {
        return Page.getInitialProps(context);
      }
      return undefined;
    }

    constructor(props, context) {
      super(props, context);
      this.pageContext = this.props.pageContext || getPageContext();
    }

    componentDidMount() {
      // Remove the server-side injected CSS.
      const jssStyles = document.querySelector('#jss-server-side');
      if (jssStyles && jssStyles.parentNode) {
        jssStyles.parentNode.removeChild(jssStyles);
      }
    }

    render() {
      return (
        <Theme>
          <JssProvider
            jss={this.pageContext.jss}
            registry={this.pageContext.sheetsRegistry}
            generateClassName={this.pageContext.generateClassName}
          >
            <MuiThemeProvider sheetsManager={this.pageContext.sheetsManager}>
              <CssBaseline />
              <Page {...this.props} />
            </MuiThemeProvider>
          </JssProvider>
        </Theme>
      );
    }
  };
}

export { Theme, withTheme as default };
