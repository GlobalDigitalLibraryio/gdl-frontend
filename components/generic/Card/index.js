/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import Card from './Card';

export default Card;
