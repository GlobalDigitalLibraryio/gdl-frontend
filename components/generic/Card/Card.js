/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import styled from 'styled-components';
import CardContent from './CardContent';
import CardImage from './CardImage';

const Card = styled.div`
  background-color: white;
  box-shadow: 0 2px 3px rgba(10, 10, 10, 0.1), 0 0 0 1px rgba(10, 10, 10, 0.1);
  color: #4a4a4a;
  max-width: 100%;
  position: relative;
`;

Card.Content = CardContent;
Card.Image = CardImage;

export default Card;
