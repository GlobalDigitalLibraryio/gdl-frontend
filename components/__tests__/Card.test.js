// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { theme } from '../../hocs/withTheme';
import Card from '../Card';

test('Renders', () => {
  const tree = shallow(<Card theme={theme}>I am in a card!</Card>);

  expect(toJson(tree)).toMatchSnapshot();
});
