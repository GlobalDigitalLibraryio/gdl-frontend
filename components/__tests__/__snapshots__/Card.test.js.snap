// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
.c0 {
  background: #fff;
  position: relative;
  max-width: 100%;
  box-shadow: 0 2px 4px rgba(0,0,0,0.1);
  color: #444;
}

<div
  className="c0"
>
  I am in a card!
</div>
`;
